client/js/jsonedit.js

```diff
@@ -1476,15 +1476,24 @@ function jeSetAndSelect(replaceBy, insideString) {
     var jsonString = JSON.stringify(jeStateNow, null, '  ');
   const startIndex = jsonString.indexOf(insideString ? '###SELECT ME###' : '"###SELECT ME###"');
   let length = jsonString.length-15-(insideString ? 0 : 2);
-
+  
+  const replaceByString = JSON.stringify(replaceBy);
   if(insideString)
-    jsonString = jsonString.replace(/###SELECT ME###/, JSON.stringify(replaceBy).substr(1, JSON.stringify(replaceBy).length-2));
+    jsonString = jsonString.replace(/###SELECT ME###/, replaceByString.substr(1, replaceByString.length-2));
   else
-    jsonString = jsonString.replace(/"###SELECT ME###"/, JSON.stringify(replaceBy));
+    jsonString = jsonString.replace(/"###SELECT ME###"/, replaceByString);
 
   jeSet(jsonString);
   const quote = typeof replaceBy == 'string' && !insideString ? 1 : 0;
-  jeSelect(startIndex + quote, startIndex+jsonString.length-length - quote, true);
+  let bracket = 0;
```

Comment:
Perhaps
```
bracket = (replaceBy && replaceByString.includes('[')) ? 1 : 0
```
and similarly for the others? The logic looks fine, though overly general (when will there ever be a wavy  bracket without a dollar sign, or both a wavy bracket and a bracket?).

Reply:
I discovered today there are other related problems when you add certain properties, so I want to rework the whole thing.